src/atomic/atm.password-input/password-input.component.tsx

```diff
@@ -0,0 +1,57 @@
+import { useState, ReactNode, forwardRef } from "react";
+import TextInput from "../atm.text-input";
+import ShowSvg from "../icons/show-svg";
+
+interface CaptionData {
+  text: string;
+  status: "error" | "success";
+}
+
+interface PasswordInputProps {
+  variant?: "disable" | "error";
```

Comment:
```suggestion
  variant?: "disabled" | "error";
```

`disable` dá a impressão de que é uma função para desabilitar

Reply:
também acho um pouco estranho que `variant` não seja um espaço para variações do campo, mas para estados dele. geralmente usamos `variant` para formas alternativas de um campo, ex: um `TextField` pode ter a forma de um campo de texto em área, um campo de senha, etc. essas seriam duas **vairantes** diferentes do campo de texto. mas uma variante pode estar em estado de erro ou desabilitada.

Reply:
entendi, então para essas funções de `disabled` e `error` por exemplo, um nome melhor seria `status` então?

Reply:
pelo oque entendi de como é usado o `variant` e `status` nos projetos, alterei esse `variant` do password e todos que usam da mesma forma(`disabled` e `error`) para `status` e fiz o mesmo para o caption, usando `status` em vez de `variants`